app/entities/user.go

```diff
@@ -47,24 +48,27 @@ func checkName(name string) error {
 
 func checkMail(email string) error {
 	const (
-		maxLocalEmailLen  int = 64
-		maxDomainEmailLen int = 255
+		maxLocalBytes  int = 64
+		maxDomainBytes int = 255
 	)
-	// Checking the total length (allowed no more than 64+1+255=320 symbols)
-	emailLen := len(email)
-	if emailLen > maxLocalEmailLen+1+maxDomainEmailLen {
-		return errors.New("email contents too many symbols: " + strconv.Itoa(emailLen))
+	// Checking the lengths of local and domain parts
+	atIndex := strings.IndexByte(email, '@')
+	if atIndex > maxLocalBytes {
+		return errors.New("local part of email contains too many bytes: " + strconv.Itoa(atIndex))
 	}
-	// Checking for some email issues by regular expression
-	if valid, _ := regexp.MatchString("(?i)"+`^([^\.@(),:;<>@[\\\]]?[\da-z!#$%&'*+,\-/=?^_\x60{|}~]+.?)+@([^.-][\da-z-]*[^.-]\.?)+([a-z]{2,6})$`, email); !valid {
```

Comment:
Could you tell me please why dont we use this approach?
```golang
const mask = `(i)^(?:[a-z\d!#$%&'*+/=?^_\x60{|}~-]+(?:\.[a-z\d!#$%&'*+/=?^_\x60{|}~-]+)*)@(?:(?:[a-z\d](?:[a-z\d-]*[a-z\d])?\.)+[a-z\d](?:[a-z\d-]*[a-z\d])?)$`
	valid, err := regexp.MatchString(mask, email)
	if err != nil {
		return fmt.Errorf("matching regex failed: %w", err)
	}
	if !valid {
		return fmt.Errorf("email does not match regex: %s", mask)
	
```

Reply:
Done it!